### PR TITLE
Parse LanguageVersion correctly

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes https://github.com/aspnet/Razor/issues/1591

### CODE REVIEW

File: test/Microsoft.AspNetCore.Mvc.Razor.Test/Internal/CSharpCompilerTest.cs [hunk 1]
```diff
@@ -151,7 +151,7 @@ public void Constructor_ConfiguresLanguageVersion()
             // Arrange
             var dependencyContextOptions = new DependencyContextCompilationOptions(
                 new[] { "MyDefine" },
-                languageVersion: "CSharp7_1",
+                languageVersion: "7.1",
```
Comment (DamianEdwards) [inline test/Microsoft.AspNetCore.Mvc.Razor.Test/Internal/CSharpCompilerTest.cs]: Is there a test that covers "latest" and "default" too?

Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.Razor.Test/Internal/CSharpCompilerTest.cs]: I could add more scenarios, but at that point, we'd be testing Roslyn's code base.

**Comment by pranavkm** [comment]
Bump?

File: src/Microsoft.AspNetCore.Mvc.Razor/Internal/CSharpCompiler.cs [hunk 2]
```diff
@@ -154,9 +154,7 @@ private void EnsureOptions()
 
             var parseOptions = new CSharpParseOptions(preprocessorSymbols: defines);
 
-            LanguageVersion languageVersion;
-            if (!string.IsNullOrEmpty(dependencyContextOptions.LanguageVersion) &&
-                Enum.TryParse(dependencyContextOptions.LanguageVersion, ignoreCase: true, result: out languageVersion))
+            if (LanguageVersionFacts.TryParse(dependencyContextOptions.LanguageVersion, out var languageVersion))
             {
                 parseOptions = parseOptions.WithLanguageVersion(languageVersion);
             }
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Mvc.Razor/Internal/CSharpCompiler.cs]: Add an empty else with a comment (or debug assert) stating that failure to parse the lang version can't really happen and why.

**Pull request closed without merging**